posts/inside-rust/2023-08-24-cargo-config-merging.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: post
+title: "Cargo changes how arrays in config are merged"
+author: Arlo Siemsen
+team: the Cargo team <https://www.rust-lang.org/governance/teams/dev-tools#cargo>
+---
+
+Cargo will be changing the order of merged configuration arrays, and we are looking for people to help test this change and provide feedback.
+
+When an array in Cargo's configuration is defined in multiple places, the elements are merged together. The order that the elements were merged was inconsistent with how other configuration types were merged together, and was not working as expected.
+
+The new merging order will match the precedence order of non-array [configuration], with higher-precedence configuration being placed later in the array. 
+
+In the case of `build.rustflags`, this resolves the confusing situation where higher precedence flags (in a project's `config.toml`) are overridden with lower precedence flags (in the global `$CARGO_HOME`, for example).
+
+This may result in behavior changes if a project depends on the existing merging order. If you have an environment that involves merging configuration arrays, please consider testing your project with nightly to ensure it will continue to work once this change stabilizes. If you encouter a problem, please file an [issue].
+
+The change is included starting with `cargo 1.74.0-nightly (2023-08-24)`.
```

Comment:
Confirmed that rust-lang/cargo@2cc50bc0b is included in `nightly-2023-08-24` toolchain.

```
$ cargo +nightly-2023-08-24 version
cargo 1.74.0-nightly (2cc50bc0b 2023-08-22)
$ cargo +nightly-2023-08-23 version
cargo 1.74.0-nightly (80eca0e58 2023-08-19)
```

However, usually the format is encoded with the commit date, like `cargo 1.74.0-nightly (<commit-sha> <commit-date>)`. The writing here is a bit confusing. I'd suggest some change like below:

```suggestion
The change is included in Cargo starting with `nightly-2023-08-24` toolchain.
```